extension_service/datastore/providers/firestore.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from datetime import datetime, timedelta
+from typing import Literal, Optional
+
+from google.cloud import firestore
+from google.cloud.firestore_v1.async_collection import AsyncCollectionReference
+from google.cloud.firestore_v1.base_query import FieldFilter
+from pydantic import BaseModel
+
+import models
+
+from .. import datastore
+
+
+class Config(BaseModel, datastore.AbstractConfig):
+    kind: Literal["firestore"]
+    projectId: Optional[str]
+
+
+class Client(datastore.Client[Config]):
+    __client: firestore.AsyncClient
+
+    @datastore.classproperty
+    def kind(cls):
+        return "firestore"
+
+    def __init__(self, client: firestore.AsyncClient):
+        self.__client = client
+
+    @classmethod
+    async def create(cls, config: Config) -> "Client":
+        return cls(firestore.AsyncClient(project=config.projectId))
+
+    async def initialize_data(
+        self,
+        airports: list[models.Airport],
+        amenities: list[models.Amenity],
+        flights: list[models.Flight],
+    ) -> None:
+        async def delete_collections(collection_list: list[AsyncCollectionReference]):
+            # Checks if colelction exists and deletes all documents
+            delete_tasks = []
+            for collection_ref in collection_list:
+                collection_exists = collection_ref.limit(1).stream()
```

Comment:
nit: do we want a bigger batch size than 1?

Reply:
We are only checking if the collection exists here, so selecting 1 document within the collection should be sufficient.